### PR TITLE
[FIX] base: js_transpiler import legacy file

### PR DESCRIPTION
**Pull request by FrancoisGe**
In a @odoo-module js file, importing a legacy file that did not have
a key respecting the "module.name" format was not supported.

This commit will support all keys that do not start with a "." or "@".

Supported:
    import X from "some/path"
    import X from "module.name"

Not supported:
    import X from "@some/path"
    import X from "./some/path"
    import X from "../../some/path"

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/tools/js_transpiler.py [hunk 1]
```diff
@@ -346,7 +346,7 @@ def convert_default_export(content):
     import\s+                           # import
     (?P<object>{(\s*\w+\s*,?\s*)+})\s*  # { a, b, c as x, ... }
     from\s*                             # from
-    (?P<path>(\".*\")|('.*')|(`.*`))    # "file path" ("some/path")
+    (?P<path>([\"'`])([^\"'`]+)(\5))    # "file path" ("some/path")
```
Comment (SimonGenin) [inline odoo/tools/js_transpiler.py:349]: Maybe try to use a name group as \5 is not readable

File: odoo/tools/js_transpiler.py [hunk 2]
```diff
@@ -400,7 +400,7 @@ def convert_legacy_default_import(content):
     import\s+                           # import
     (?P<identifier>\w+)\s*              # default variable name
     from\s*                             # from
-    (?P<path>(\".*\")|('.*')|(`.*`))    # "file path" ("some/path")
+    (?P<path>([\"'`])([^\"'`]+)(\4))    # "file path" ("some/path")
```
Comment (SimonGenin) [inline odoo/tools/js_transpiler.py:403]: Same with \4

File: odoo/addons/test_assetsbundle/tests/test_js_transpiler.py [hunk 3]
```diff
@@ -247,15 +247,16 @@ def test_06_import(self):
 
 import Line9  from "test.Dialog";
 import  { Line10, Notification }  from 'test.Dialog2';
+import Line11  from "test/Dialog";
 
-import * as Line11 from "test.Dialog";
+import * as Line12 from "test.Dialog";
```
Comment (SimonGenin) [inline odoo/addons/test_assetsbundle/tests/test_js_transpiler.py:252]: Probably should add a test instance with longer names (multiple dots) as it was a failing case as well.

File: odoo/tools/js_transpiler.py [hunk 4]
```diff
@@ -369,11 +369,11 @@ def repl(matchobj):
 
 IMPORT_LEGACY_DEFAULT_RE = re.compile(r"""
     ^
-    (?P<space>\s*)                                      # space and empty line
-    import\s+                                           # import
-    (?P<identifier>\w+)\s*                              # default variable name
-    from\s*                                             # from
```
Comment (xmo-odoo) [inline odoo/tools/js_transpiler.py:375]: ... don't reindent everything, that's completely unnecessary diff garbage.

File: odoo/addons/test_assetsbundle/tests/test_js_transpiler.py (hunk 3 above)
Comment (xmo-odoo) [inline odoo/addons/test_assetsbundle/tests/test_js_transpiler.py:252]: ... maybe don't use a sequence if that requires this sort of completely useless edits? Just call the new line something else, or add it at the end.

File: odoo/addons/test_assetsbundle/tests/test_js_transpiler.py [hunk 5]
```diff
@@ -247,15 +247,16 @@ def test_06_import(self):
 
 import Line9  from "test.Dialog";
 import  { Line10, Notification }  from 'test.Dialog2';
+import Line11  from "test/Dialog";
```
Comment (xmo-odoo) [inline odoo/addons/test_assetsbundle/tests/test_js_transpiler.py:250]: How does this differ from `Line7` and `Line8`?

File: odoo/addons/test_assetsbundle/tests/test_js_transpiler.py [hunk 6]
```diff
@@ -255,7 +255,10 @@ def test_06_import(self):
 import {Line13}  from "@test.Dialog" //HELLO
 
 
-const test = `import { Line13, Notification } from "../src/Dialog";`
+const test = `import { Line14, Notification } from "../src/Dialog";`
+
+import Line15 from "test/Dialog";
+import Line16 from "test.Dialog.error";
```
Comment (xmo-odoo) [inline odoo/addons/test_assetsbundle/tests/test_js_transpiler.py:261]: Could the commit message explain more clearly why / how these differ from cases 7/8/9?

File: odoo/tools/js_transpiler.py [hunk 7]
```diff
@@ -251,11 +251,11 @@ def repl(matchobj):
...
     ^
-    (?P<space>\s*)                      # space and empty line
-    export\s*                           # export
-    (?P<object>{(\s*\w+\s*,?\s*)*})\s*  # { a, b, c as x, ... }
-    from\s*                             # from
-    (?P<path>(\".*\")|('.*')|(`.*`))    # "file path" ("some/path.js")
+    (?P<space>\s*)                                  # space and empty line
+    export\s*                                       # export
+    (?P<object>{(\s*\w+\s*,?\s*)*})\s*              # { a, b, c as x, ... }
+    from\s*                                         # from
```
Comment (xmo-odoo) [inline odoo/tools/js_transpiler.py:257]: These lines should not change, same for the rest.

File: odoo/addons/test_assetsbundle/tests/test_js_transpiler.py (hunk 6 above)
Comment (FrancoisGe) [reply odoo/addons/test_assetsbundle/tests/test_js_transpiler.py:261]: 15 and 16 are legacy js file imports.
7/8 and 9 are @odoo-module js file imports

File: odoo/tools/js_transpiler.py (hunk 7 above)
Comment (SimonGenin) [reply odoo/tools/js_transpiler.py:257]: It is annoying for sure but
1) We really want to keep the code nice as this file is in its first iterations (so we want a good indentation)
2) Technically, this whole thing is one regex, so the diff is not that shocking
3) fge and I wrote this feature, so no blame get lost
With that, we believe it makes sense to keep it

Comment (xmo-odoo) [reply odoo/tools/js_transpiler.py]: Move the comments to be above the lines they're commenting on then. This style of commenting is horrible for exactly the reason being demonstrated here.

**Comment by aab-odoo** [comment]
robodoo r+
